packages/kbn-test/src/page_load_metrics/navigation.ts

```diff
@@ -0,0 +1,147 @@
+    protocol: baseUrl.protocol,
+    hostname: baseUrl.hostname,
+    port: baseUrl.port,
+    pathname: joinPath(baseUrl.pathname || '', path),
+  });
+}
+
+async function loginToKibana(
+  log: ToolingLog,
+  browser: puppeteer.Browser,
+  options: NavigationOptions
+) {
+  log.debug(`log in to the app..`);
+  const page = await browser.newPage();
+  const loginUrl = createUrl('/login', options);
+  await page.goto(loginUrl, {
+    waitUntil: 'networkidle0',
+  });
+  await page.type('[data-test-subj="loginUsername"]', options.appConfig.username);
+  await page.type('[data-test-subj="loginPassword"]', options.appConfig.password);
+  await page.click('[data-test-subj="loginSubmit"]');
+  await page.waitForNavigation({ waitUntil: 'networkidle0' });
+  await page.close();
+}
+
+export async function navigateToApps(log: ToolingLog, options: NavigationOptions) {
+  const browser = await puppeteer.launch({ headless: options.headless, args: ['--no-sandbox'] });
+  const devToolsResponses: NavigationResults = new Map();
+  const apps = [
+    { path: '/app/discover', locator: '[data-test-subj="discover-sidebar"]' },
+    { path: '/app/home', locator: '[data-test-subj="homeApp"]' },
+    { path: '/app/canvas', locator: '[data-test-subj="create-workpad-button"]' },
+    { path: '/app/maps', locator: '[data-test-subj="newMapLink"]' },
+    { path: '/app/apm', locator: '[data-test-subj="apmMainContainer"]' },
+  ];
+
+  await loginToKibana(log, browser, options);
+
+  await Promise.all(
+    apps.map(async (app) => {
+      const page = await browser.newPage();
+      page.setCacheEnabled(false);
+      page.setDefaultNavigationTimeout(0);
+      const frameResponses = new Map<string, FrameResponse>();
+      devToolsResponses.set(app.path, frameResponses);
+
+      const client = await page.target().createCDPSession();
+      await client.send('Network.enable');
+
+      function getRequestData(requestId: string) {
+        if (!frameResponses.has(requestId)) {
+          frameResponses.set(requestId, { url: '', dataLength: 0 });
+        }
+
+        return frameResponses.get(requestId)!;
+      }
+
+      client.on('Network.responseReceived', (event: ResponseReceivedEvent) => {
+        getRequestData(event.requestId).url = event.response.url;
+      });
+
+      client.on('Network.dataReceived', (event: DataReceivedEvent) => {
+        getRequestData(event.requestId).dataLength += event.dataLength;
+      });
+
+      const url = createUrl(app.path, options);
+      log.debug(`goto ${url}`);
+      await page.goto(url, {
+        waitUntil: 'networkidle0',
+      });
+
+      let readyAttempt = 0;
+      let selectorFound = false;
+      while (!selectorFound) {
+        readyAttempt += 1;
+        try {
+          await page.waitForSelector(app.locator, { timeout: 5000 });
+          selectorFound = true;
+        } catch (error) {
+          log.error(
+            `Page '${app.path}' was not loaded properly, unable to find '${
```

Comment:
shouldn't we log `error` after https://github.com/elastic/kibana/pull/66224/files#diff-12d35330911b009e894d60073c6466ccR132 ? Or we should use another logging level here

Reply:
I think it is on purpose to show that we are retrying operation

Reply:
```
 debg log in to the app..
 debg goto http://localhost:5620/app/discover
 debg goto http://localhost:5620/app/home
 debg goto http://localhost:5620/app/maps
 debg goto http://localhost:5620/app/canvas
 debg goto http://localhost:5620/app/apm
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR Page '/app/discover' was not loaded properly, unable to find '[data-test-subj="discover-sidebar"]', url: http://localhost:5620/app/kibana
ERROR UNHANDLED ERROR
ERROR TimeoutError: waiting for selector "[data-test-subj="discover-sidebar"]" failed: timeout 5000ms exceeded
          at new WaitTask (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/DOMWorld.js:388:34)
          at DOMWorld._waitForSelectorOrXPath (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/DOMWorld.js:313:26)
          at DOMWorld.waitForSelector (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/DOMWorld.js:296:21)
          at Frame.waitForSelector (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/FrameManager.js:384:51)
          at Frame.<anonymous> (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/helper.js:95:27)
          at Page.waitForSelector (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/node_modules/puppeteer/lib/Page.js:799:33)
          at Promise.all.apps.map (/Users/dzmitrylemechko/github/kibana/packages/kbn-test/src/page_load_metrics/navigation.ts:123:22)
```